datasets/ba/companies/crawler.py

```diff
@@ -0,0 +1,522 @@
+            entity.add("status", record["status_bankruptcy"], lang="bos")
+
+        entity.add("country", "ba")
+        entity.add("address", record["address"], lang="bos")
+        if record["address_additional"]:
+            entity.add("address", record["address_additional"], lang="bos")
+
+        entity.add("legalForm", record["legal_form"], lang="bos")
+        entity.add("registrationNumber", record["registration_number"])
+
+        # JD: LegalEntity:idNumber probably?
+        if record.get("unique_id"):
+            entity.add("registrationNumber", record["unique_id"])
+
+        entity.add("sourceUrl", record["details_url"])
+        entity.add("modifiedAt", record["date_of_last_decision"])
+        entity.add("retrievedAt", datetime.datetime.now().isoformat())
+        context.emit(entity, target=True)
+
+        for person in founders_people:
+            founder = context.make("Person")
+            founder.id = context.make_id("BAFounder", entity.id, person["name"])
+            founder.add("name", person["name"], lang="bos")
+            context.emit(founder)
+
+            own = context.make("Ownership")
+            own.id = context.make_id("BAOwnership", entity.id, founder.id)
+            own.add("asset", entity.id)
+            own.add("owner", founder.id)
+            own.add("ownershipType", "Founder", lang="eng")
+
+            # JD: Not sure where to map shares and capital_contracted
+            own.add("sharesValue", person["capital_paid"])
+            context.emit(own)
+
+        for comp in founders_companies:
+            founder_company = context.make("LegalEntity")
+            founder_company.id = context.make_id(
+                "BAFounderCompany", entity.id, comp["name"]
+            )
+            founder_company.add("name", comp["name"], lang="bos")
+            if comp.get("country"):
+                founder_company.add("country", comp["country"], lang="bos")
+
+            if comp.get("address"):
+                founder_company.add("address", comp["address"], lang="bos")
+
+            if comp.get("registration_number"):
+                founder_company.add("registrationNumber", comp["registration_number"])
+
+            context.emit(founder_company)
+
+            own = context.make("Ownership")
+            own.id = context.make_id("BAOwnership", entity.id, founder_company.id)
+            own.add("asset", entity.id)
+            own.add("owner", founder_company.id)
+            own.add("ownershipType", "Founder", lang="eng")
+
+            context.emit(own)
+
+        for man in managers:
+            director = context.make("Person")
+            director.id = context.make_id("BAdirector", entity.id, man["name"])
+            director.add("name", man["name"], lang="bos")
+            context.emit(director)
+
+            rel = context.make("Directorship")
+            rel.id = context.make_id("BADirectorship", entity.id, director.id)
+            rel.add("role", man["position"], lang="bos")
```

Comment:
is position ever anything that might make us pick  a different schema? if so, we tend to do that using lookups, then there's no need to add it to :role.

Reply:
Sorry, I don't understand this part and how it works. `lookups` is not really described in the docs, I fear.